src/ProjectSystem/Impl/IO/MsBuildFileSystemWatcher.cs

```diff
@@ -215,6 +215,9 @@ public sealed partial class MsBuildFileSystemWatcher : IDisposable {
 
             relativePath = PathHelper.MakeRelative(rootDirectory, fullPath);
             try {
+                if (!fileSystem.FileExists(fullPath)) {
```

Comment:
Hm... then the first case requires further investigation. Any `IFileSystemChange` appears in queue only after event has happened. So if `FileCreated` happened but no file was found, there should be another `FileCreated` in the queue that should add a file. Or there should be an error in the log.
Do you have logs for the #2070?